### PR TITLE
guid/luid conversion functionallity and pkthdr is writable

### PR DESCRIPTION
**Pull request by finsor**
Added guid/luid conversion functionality as independent functions and in open_live. 
Also added pkthdr.setlen/setcaplen/addlen/addcaplen , so pkthdr can be writable.

I needed these functionalities in my project, so I added them. Sorry if the code doesn't look good :)

### CODE REVIEW

File: pcapy.cc [hunk 1]
```diff
@@ -274,3 +309,274 @@ initpcapy(void)
...
+
+
+
+
+
+
+
+
+
+
```
Comment (finsor) [inline pcapy.cc]: * from wireshark's source *

**Comment by martinuy** [comment]
Hi @finsor ,

Thanks for your contributions! We really appreciate that :-)

I'd like to understand the use-case to include packet header modification APIs. Can you please elaborate more on that? Even if we agree that there is good reason for it, I would suggest proposing more generic APIs instead of one to modify each property in a certain way (i.e. length addition is too specific for an API).

In regards to the luid_to_guid feature, does it depend on a pcap on some way? Because it looks like an auxiliary function, not necessarily related.

Kind regards,
Martin.-

**Comment by martinuy** [comment]
Hi @finsor 

Apologies for the delay. I do this in my free time and split my time with other projects too.

I understand your use-case but I'm still not convinced of adding packet header modification to pcapy's API. Sorry about that. If your use-case turns to be a common and requested use-case, we may revisit this decision in the future.

**Comment by jmelchiori** [comment]
I'd like to second the need for a header modification API.

My use case would be modifying the timestamp before dumping the packets into a new file.  Specifically replacing the os-provided timestamp with one written by a tap-aggregator (ie. Arista Tapagg, cPacket, Metamako, etc) to the fcs or another packet trailer format.

**Pull request closed without merging**